### PR TITLE
[WW-5073] - simplify AbstractMatcher convertParam

### PR DESCRIPTION
**Pull request by sdutry**
[WW-5073](https://issues.apache.org/jira/browse/WW-5073)

### CODE REVIEW

File: core/src/main/java/com/opensymphony/xwork2/config/impl/AbstractMatcher.java [hunk 1]
```diff
@@ -204,31 +211,22 @@ public E match(String potentialMatch) {
...
-        }
-        
-        return ret.toString();
-    }
+	protected String convertParam(String val, Map<String, String> vars) {
+		if (val == null) {
+			return null;
+		}
+
+		Pattern wildcardPattern = Pattern.compile("\\{(.)\\}");
```
Comment (lukaszlenart) [inline core/src/main/java/com/opensymphony/xwork2/config/impl/AbstractMatcher.java:219]: Just a small optimisation, could you move this pattern compilation into a field? Right now each time the `convertParam` is called, the same pattern is compiled over and over.

**Review by lukaszlenart — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/opensymphony/xwork2/config/impl/AbstractMatcher.java (hunk 1 above)
Comment (sdutry) [reply core/src/main/java/com/opensymphony/xwork2/config/impl/AbstractMatcher.java]: Moved.

Comment (lukaszlenart) [reply core/src/main/java/com/opensymphony/xwork2/config/impl/AbstractMatcher.java]: 👏

**Review by lukaszlenart — APPROVED** [review]
LGTM 👍

**Pull request merged**